alert-database/src/main/java/com/synopsys/integration/alert/database/api/DefaultNotificationManager.java

```diff
@@ -57,17 +61,27 @@
     private final NotificationContentRepository notificationContentRepository;
     private final AuditEntryRepository auditEntryRepository;
     private final AuditNotificationRepository auditNotificationRepository;
+    private final EventManager eventManager;
 
     @Autowired
-    public DefaultNotificationManager(final NotificationContentRepository notificationContentRepository, final AuditEntryRepository auditEntryRepository, final AuditNotificationRepository auditNotificationRepository) {
+    public DefaultNotificationManager(final NotificationContentRepository notificationContentRepository, final AuditEntryRepository auditEntryRepository, final AuditNotificationRepository auditNotificationRepository,
+        @Lazy final EventManager eventManager) {
         this.notificationContentRepository = notificationContentRepository;
         this.auditEntryRepository = auditEntryRepository;
         this.auditNotificationRepository = auditNotificationRepository;
+        this.eventManager = eventManager;
     }
 
     @Override
-    public AlertNotificationWrapper saveNotification(final AlertNotificationWrapper notification) {
-        return notificationContentRepository.save((NotificationContent) notification);
+    public List<AlertNotificationWrapper> saveAllNotifications(final Collection<AlertNotificationWrapper> notifications) {
+        final List<AlertNotificationWrapper> notificationContents = notifications.stream()
+                                                                        .map(notification -> notificationContentRepository.save((NotificationContent) notification))
+                                                                        .collect(Collectors.toList());
+        final List<Long> notificationIds = notificationContents.stream()
+                                               .map(AlertNotificationWrapper::getId)
+                                               .collect(Collectors.toList());
+        eventManager.sendEvent(new NotificationEvent(notificationIds));
```

Comment:
I feel that this functionality doesn't exactly fit here. We should do our save then if people want, the user can send the event. This would also help remove the circular dependency caused by Autowiring the EventManager.

Reply:
The problem is, if we don't do it here then anywhere we call saveAllNotifications needs access to the EventManager AND they would have this same duplicated code
`
 final List<Long> notificationIds = notificationContents.stream()
                                               .map(AlertNotificationWrapper::getId)
                                               .collect(Collectors.toList());
        eventManager.sendEvent(new NotificationEvent(notificationIds));
`